File: sstockjson.py
# import json parsing and http get requests packages

import json
import requests #http get/post library, installed as Python packages via separate step
import sys
import re # regular expression matching including http link matching

# sample usage: ./sstocksearch.py [images or videos or audio] [keyword]
# syntax: sstocksearch.py  media-type keyword (search for micro stock media-type content based on keyword)

print "starting shutterstock API"
S1="https://api.shutterstock.com/v2/"
S2=sys.argv[1]
S3="/search?per_page=15&query="
S4=sys.argv[2]
S5="&view=full"
S6=S1 + S2 + S3 + S4 + S5
print "shutterstock api url %s" % (S6)

def byteify(input):
    if isinstance(input, dict):
        return {byteify(key):byteify(value) for key,value in input.iteritems()}
    elif isinstance(input, list):
        return [byteify(element) for element in input]
    elif isinstance(input, unicode):
        return input.encode('utf-8')
    else:
        return input

# set the url for the video search along with the basic authorization

if (S6 != ""):
    url = S6
else:
    url='https://api.shutterstock.com/v2/videos/search?per_page=10&query=ocean&view=full'
    
print "url = %s" % (url)

head={'Authorization': 'Basic MmY3MTAwNjdmNWIwNGY0NDNmZWE6NWRiM2QyODYxZDQ4ZDlmZmU0YzcxMzQ5NDQ4MjM5MDBlZmYzNDU2MA=='}

# make an http request (like curl, but here in Python) to the video search url using the basic authentication

r=requests.get(url,headers=head)
r.text

# load the json data from the http data stream returned from the http url

decoded = byteify(json.loads(r.text))

# print out json data as needed

print decoded['page']
print decoded['per_page']

# loop over video search items and print out url etc.

for x in range(0, decoded['per_page']):
  if (sys.argv[1] == 'videos'):
    print "Video %s url = %s" % (x, decoded['data'][x]['assets']['preview_mp4']['url'])
  elif (sys.argv[1] == 'audio'):
    print "Audio %s url = %s" % (x, decoded['data'][x]['assets']['preview_mp3']['url'])
  elif (sys.argv[1] == 'images'):
    print "Image %s url = %s" % (x, decoded['data'][x]['assets']['preview']['url'])

    

    

File: sstockjson.pl
#!/usr/bin/perl
# import http::request perl package, installed from cpan via install http::request

use strict;
use LWP::UserAgent;
use Encode qw(decode encode);

# sample usage: ./sstockjson.pl [images or videos or audio] [keyword]
# syntax: sstockjson.pl  media-type keyword (search for micro stock media-type content based on keyword)

print "starting shutterstock API\n";
my $s1="https://api.shutterstock.com/v2/";
my $s2=$ARGV[0];
my $s3="/search?per_page=1&query=";
my $s4=$ARGV[1];
my $s5="&view=full";
my $s6=$s1 . $s2 . $s3 . $s4 . $s5;
print "shutterstock api url\n", $s6, "\n";

# initialize the user agent for the http request to query for media content

my $ua = LWP::UserAgent->new;

# set the url for the http request

my $server_endpoint = $s6;
 
# set the custom HTTP request header fields for content-type and basic authorization

my $req = HTTP::Request->new(GET => $server_endpoint);
#$req->header('content-type' => 'application/json');
$req->default_header(Authorization => "Basic " . "MmY3MTAwNjdmNWIwNGY0NDNmZWE6NWRiM2QyODYxZDQ4ZDlmZmU0YzcxMzQ5NDQ4MjM5MDBlZmYzNDU2MA==");

print ("http basic authorization request", $req->default_header('Authorization'), "\n");

# make the http request with header fields and print the successful or failed response 

my $resp = $ua->request($req);
if ($resp->is_success) {
    my $message = $resp->decoded_content;
    print "Received reply: $message\n";
    # encode the data as utf-8 and print it
    my $messageutf8 = encode_utf8($message);
    print "utf-8 encoded message $message\n";
}
else {
    print "HTTP GET error code: ", $resp->code, "\n";
    print "HTTP GET error message: ", $resp->message, "\n";
}


